lib/bundler/cli/gem.rb

```diff
@@ -29,6 +29,11 @@ def run
       constant_name = name.gsub(/-[_-]*(?![_-]|$)/) { "::" }.gsub(/([_-]+|(::)|^)(.|$)/) { $2.to_s + $3.upcase }
       constant_array = constant_name.split("::")
 
+      if !Bundler.git_present?
+        Bundler.ui.error "You need to install git to be able to use bundle gem."
```

Comment:
`git ` is needed only for fetching `username` `user_email` `git_author_name`, to write them in generated` mygem.gemspec` file and `README.md` file, which can be edited by the user afterward. 

So now I think it must not stop generating files (When git is not installed then just replace `username`, `email` with `["TODO: Write your name/email"]` in generated `mygem.gemspec` file)